### PR TITLE
[PROD] :sparkles: filter events already finished on splash page

### PR DESCRIPTION
**Pull request by antoineeudes**

### CODE REVIEW

File: src/components/venues/Jazzbar/EntranceExperience/EntranceExperience.tsx [hunk 1]
```diff
@@ -74,16 +75,33 @@ const JazzbarEntranceExperience: React.FunctionComponent<PropsType> = ({
     purchaseHistoryRequestStatus: boolean;
   };
 
+  const futureOrOngoingVenueEvents =
```
Comment (maxb94) [inline src/components/venues/Jazzbar/EntranceExperience/EntranceExperience.tsx:78]: venueEvents?.filter(...

File: src/components/venues/Jazzbar/EntranceExperience/EntranceExperience.tsx [hunk 2]
```diff
@@ -74,16 +75,33 @@ const JazzbarEntranceExperience: React.FunctionComponent<PropsType> = ({
...
 
   useEffect(() => {
-    if (user && venueEvents && venueId && eventId && redirectTo === "payment") {
-      setSelectedEvent(venueEvents.find((event) => event.id === eventId));
+    if (
+      user &&
+      futureOrOngoingVenueEvents &&
+      venueId &&
+      eventId &&
+      redirectTo === "payment"
```
Comment (maxb94) [inline src/components/venues/Jazzbar/EntranceExperience/EntranceExperience.tsx:97]: this is quite hard to parse. Could you create a variable with a nice name e.g. const eventRequiringPayment = ...

File: src/components/venues/Jazzbar/EntranceExperience/EntranceExperience.tsx [hunk 3]
```diff
@@ -74,16 +75,33 @@ const JazzbarEntranceExperience: React.FunctionComponent<PropsType> = ({
...
+    ) {
+      setSelectedEvent(
+        futureOrOngoingVenueEvents.find((event) => event.id === eventId)
```
Comment (maxb94) [inline src/components/venues/Jazzbar/EntranceExperience/EntranceExperience.tsx:100]: what happens if this is undefined?

File: src/components/venues/Jazzbar/EntranceExperience/EntranceExperience.tsx [hunk 4]
```diff
@@ -206,99 +224,111 @@ const JazzbarEntranceExperience: React.FunctionComponent<PropsType> = ({
...
+                                </div>
+                              ) : (
+                                <div className="price-container">
+                                  Individual tickets £{venueEvent.price / 100}
+                                  <br />
+                                  Group tickets £
+                                  {venueEvent.collective_price / 100}
+                                  {!user && (
+                                    <div className="login-invitation">
+                                      Already have a ticket?{" "}
```
Comment (maxb94) [inline src/components/venues/Jazzbar/EntranceExperience/EntranceExperience.tsx:292]: remove the {" "} and use {\`Already have a ticket? \`}

File: src/components/venues/Jazzbar/EntranceExperience/EntranceExperience.tsx (hunk 2 above)
Comment (antoineeudes) [reply src/components/venues/Jazzbar/EntranceExperience/EntranceExperience.tsx:97]: this code isn't usefule anymore as we use modals for the authentication
Thanks!

File: src/components/venues/Jazzbar/EntranceExperience/EntranceExperience.tsx (hunk 4 above)
Comment (thecadams) [reply src/components/venues/Jazzbar/EntranceExperience/EntranceExperience.tsx:292]: Annoyingly, the linter we are using (`npm run lint`) likes to do the {" "} so it can split JSX strings across lines. Whichever we go with it'd be great to immortalise it in a lint rule.

File: src/components/venues/Jazzbar/EntranceExperience/EntranceExperience.tsx (hunk 3 above)
Comment (thecadams) [reply src/components/venues/Jazzbar/EntranceExperience/EntranceExperience.tsx]: aye. A possible mitigation is to change its type to something non-nullable to prevent this being a problem.

File: src/components/venues/Jazzbar/EntranceExperience/EntranceExperience.tsx (hunk 4 above)
Comment (maxb94) [reply src/components/venues/Jazzbar/EntranceExperience/EntranceExperience.tsx:271]: ah that's annoying :( agree we should fix it at some point
